### ISSUE TITLE
Azure EventHub binding is not properly check pointing messages / processing old messages.

### ISSUE THREAD

**Issue by MattCosturos** [dapr/components-contrib#1598]
I have an Azure IoT Hub routing message to an Azure Event Hub. The event hub has a TTL of 1 day.
I have a Event Hub binding processing messages in my Telemetry Service.
I create an edge device and begin sending messages at T=0:00.
I start my Telemetry Service at T+5 minutes.

## Expected Behavior
I expect the dapr binding to retrieve all messages in the event hub that have not been retrieved yet. 
AKA I expect to get messages from the device starting at T=0.

## Actual Behavior
The dapr binding only retrieves messages that come in AFTER the component was started up.

## Steps to Reproduce the Problem
Create Azure IoT Hub.
Create Azure Event hub.
Route messages from iot hub to Event hub. 
Create a .net service to consume the messages using Azure Event Hub binding.
Create an Edge device that generates iot messages.
Start the edge device and confirm it is sending messages.
Start the service to consume the messages.
Confirm that it only processed new messages that were generated after the service was started

## Release Note

RELEASE NOTE:

**Comment by berndverst** [comment]
Are there other clients for the same consumer group? Please ensure you are using a dedicated consumer group for use with Dapr.

https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features#checkpointing

**Comment by MattCosturos** [comment]
I will double check that the consumer groups are unique to dapr. 
Related follow up question.
My event hub has 2 partitions. I have created 2 bindings, and specified which partition each binding will connect to.
Does each binding need it's own consumer group? or is the single consumer group ok where the binding are reading from the same event hub but different partitions?

I remember check pointing working when we first set this up, but it broke at some point, possibly when I added the second binding

**Comment by MattCosturos** [comment]
Create brand new consumer group on my event hub.
Setup components to use new group. (2 event hub bindings, for Partition 0 and partition 1)
Startup all my actor service and telemetry processing service.
Start up edge device that sends messages. Confirm messages come in to Telemetry processing service.
Stop Telemetry Processing Service.
Wait 1 minute.
Stop device that sends messages.
Start Telemetry Processing Service.
No additional messages arrived from event hub binding.

Test 2.
Start services.
Start up edge device.
Run for 2 minutes (confirmed messages arrived).
Stop telemetry processing service. Made note of last sequence number message to arrive (1733)
Wait 2 minutes
Start telemetry service. Message with sequence number 3157 arrived.
My timing was not exact, closer to 2:22 between stopping and restarting, and we missed that amount of messages.

![image](https://user-images.githubusercontent.com/48531957/160399066-f5a74a97-02df-4351-a875-697c28de3cf2.png)

Attached file shows event hub metrics. Incoming matches my expected rate.
Outgoing only sends when my service is running WHILE messages are coming in.

What more info about my setup to I need to provide?

**Comment by berndverst** [comment]
@MattCosturos apologies for the delay - we have been 100% occupied trying to get release 1.7.0 out of the door (hopefully very soon).

Are you using the same storage account container or separate storage account containers? That's another thing I would recommend trying. Using the same storage account itself should be fine, but you should use distinct storage containers for each subscriber.

**Comment by MattCosturos** [comment]
I do have a unique storage account container using used, but a common account.
To review my binding settings.
I have a Connection string pointing to the event hub.
I have a shared Storage Account Name, and  Storage Account Key.
I have a unique Storage Container Name.
I have a unique Consumer Group.

**Comment by MattCosturos** [comment]
This is not resolved.
I'm not sure exactly what I need to provide.
I have created brand new consumer groups to ensure no other consumer is using the same group.
My dapr event hub bindings only consume event hub messages WHEN running.

**Comment by artursouza** [comment]
I agree that the component should support starting from the beginning of the topic but this behavior should be configurable instead. Some scenarios might require to start at the end of the topic.

**Comment by MattCosturos** [comment]
Did the behavior change at some point? When we first started testing with dapr, the behavior was to resume processing at the last received message. I'm not quite sure when it broke.

**Comment by artursouza** [comment]
> Did the behavior change at some point? When we first started testing with dapr, the behavior was to resume processing at the last received message. I'm not quite sure when it broke.

Interesting point, we should not break that behavior.

**Comment by MattCosturos** [comment]
Any update on this? Any more info I can provide?

Just so I'm clear, right now, if an event hub binding is not listening, those messages will not be processed?
The checkpointing feature does not work, and that is by design at this point?

**Comment by MattCosturos** [comment]
One more reply with some helpful info.

I've been doing some testing, and assigning a event hub binding to a specific partition seems to break check pointing.

All tests are using an Event hub with 2 partitions.
1 dapr event hub binding, 1 consumer group - Check pointing works.
2 dapr bindings, 1 consumer group, both bindings floating - Check pointing works.
2 dapr bindings, 1 consumer group, each binding fixed to a specific partition - Check pointing breaks!
2 dapr bindings, 2 different consumer groups, each binding fixed to a specific partition - Check pointing breaks!
2 dapr bindings, 2 different consumer groups, both bindings floating - Check pointing works.


So if we wish to have an event hub with multiple partitions, and process messages in parallel, we need to create multiple bindings, but we need to leave each binding floating instead of assigning it to a specific partition?

**Comment by yaron2** [comment]
cc @ItalyPaleAle

**Comment by MattCosturos** [comment]
One more follow up message.
I can confirm that if you add a partition Id to an event hub binding, then the storage account container will no longer be 
updated with check pointing info, so of course check pointing will break.

If this is by design, it needs to be documented here.
https://docs.dapr.io/reference/components-reference/supported-bindings/eventhubs/

**Comment by tanvigour** [comment]
Hi @MattCosturos , here's the certification test https://github.com/dapr/components-contrib/blob/master/tests/certification/bindings/azure/eventhubs/eventhubs_test.go#L330
which covers a case with 2 dapr bindings where each binding is fixed to a partition id and that seems to go well. Is this use case that breaks for you?

**Comment by berndverst** [comment]
The "expected" behavior described is not necessarily the behavior everyone will want. This item will require more discussion and investigation and will not make the 1.10 release.

**Comment by MattCosturos** [comment]
I don't know what should be considered right, or wrong or what should be the Default behavior, BUT I know that the behavior needs to be documented clearly. And the options to turn check pointing on / off needs to exist, and be documented clearly.

Otherwise one can't rely on Dapr delivering all messages.

**Comment by berndverst** [comment]
cc @ItalyPaleAle

**Comment by berndverst** [comment]
Can you verify the behavior against Dapr 1.10 @MattCosturos ? Checkpointing works differently now and we rewrote the code to use a completely different SDK.

**Comment by MattCosturos** [comment]
It's been a while since I looked at this.

The option that caused me problems appears to not even exist anymore in the binding. Per [docs](https://docs.dapr.io/reference/components-reference/supported-bindings/eventhubs/) I don't even see the option to explicitly set a binding to use a specific partition.

Our current setup has an event hub with N partitions and our microservice uses N bindings. None of the bindings call out a specific partition, and the checkpointing appears to work.

I just did a test of 1.10.0 and the checkpoint works as  expected. 1 event hub, 2 partitions, 2 bindings.
Stopped service, wait, restart service, dapr binding resumes processing messages based on checkpointing.

In the past the default behavior changed or broke when I set a partition on the binding, but as it stands, everything works the way I need it.

The only issue I really see is the documentation page does not mention checkpointing at all.
